### PR TITLE
Appveyor package

### PR DESCRIPTION
**Pull request by maximedenes**
This PR fixes some issues in the Windows build script and makes AppVeyor generate our Windows package automatically.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -12,15 +12,38 @@ environment:
   CYGMIRROR: http://ftp.inf.tu-dresden.de/software/windows/cygwin32
   CYGCACHE: C:\cygwin64\var\cache\setup
   opam_url: https://github.com/fdopen/opam-repository-mingw/releases/download/0.0.0.1/opam64.tar.xz
+  DESTCOQ: C:\coq64_inst
+  matrix:
+  - USEBUILDSCRIPT: false
+  - USEBUILDSCRIPT: true
```
Comment (Zimmi48) [inline appveyor.yml]: `USEBUILDSCRIPT: false` is for running the test-suite with OPAM
`USEBUILDSCRIPT: true` is for building the Coq packages
Given how convoluted this is, short comments giving this info in the appveyor file wouldn't hurt (e.g. on the two lines above).

File: appveyor.yml [hunk 2]
```diff
@@ -12,15 +12,38 @@ environment:
...
 
 install:
-- cmd: '%CYGROOT%\setup-x86_64.exe -qnNdO -R %CYGROOT% -l %CYGCACHE% -s
-  %CYGMIRROR% -P rsync -P patch -P diffutils -P make -P unzip -P m4 -P findutils -P time'
-- cmd: '%CYGROOT%/bin/bash -l %APPVEYOR_BUILD_FOLDER%/dev/build/windows/appveyor.sh'
+- cmd: 'if %USEBUILDSCRIPT% == false (%CYGROOT%\setup-x86_64.exe -qnNdO -R %CYGROOT% -l %CYGCACHE% -s
+  %CYGMIRROR% -P rsync -P patch -P diffutils -P make -P unzip -P m4 -P findutils -P time)'
+- cmd: 'if %USEBUILDSCRIPT% == false (%CYGROOT%/bin/bash -l -c "/usr/bin/timeout -s SIGTERM -k 1s 30.0s curl --insecure --retry 2 --head \"https://github.com\"")'
+- cmd: 'if %USEBUILDSCRIPT% == false (%CYGROOT%/bin/bash -l %APPVEYOR_BUILD_FOLDER%/dev/build/windows/appveyor.sh)'
```
Comment (Zimmi48) [inline appveyor.yml]: In passing, let me remark that I don't really get how it makes sense to have the `appveryor.sh` script live in `dev/build/windows/` and not instead in `dev/ci`.

File: dev/build/windows/makecoq_mingw.sh [hunk 3]
```diff
@@ -1058,13 +1058,18 @@ function copq_coq_gtk {
     install_rec  "$PREFIX/share/themes/" '*'                          "$PREFIXCOQ/share/themes"
     
     # This below item look like a bug in make install
+    if [ -d "$PREFIXCOQ/share/coq/" ] ; then
+      COQSHARE="$PREFIXCOQ/share/coq/"
+    else
+      COQSHARE="$PREFIXCOQ/share/"
+    fi
     if [[ ! $COQ_VERSION == 8.4* ]] ; then
```
Comment (Zimmi48) [inline dev/build/windows/makecoq_mingw.sh]: Does it make sense to keep these tests? It's not like we are ever going to tests these scripts with 8.4 again.

**Comment by maximedenes** [comment]
@Zimmi48 Thanks for the comments, but this PR is tagged "in progress", so I think it is a bit early to review ;)

**Comment by Zimmi48** [comment]
I was not expecting you would leave any time frame to review the PR once you remove the `needs: progress` flag (^^).

**Comment by Zimmi48** [comment]
Removing the `needs: progress` flag since you are telling me that this is ready to backport. Note that I don't like this too much and will prepare the backport and do the testing but wait until the PR is actually merged before actually backporting. If you are changing anything between now and the merge, please alert me.

**Pull request merged**